engine/src/main/scala/cromwell/jobstore/JobStoreReader.scala

```diff
@@ -1,28 +1,36 @@
 package cromwell.jobstore
 
-import akka.actor.{Actor, ActorLogging}
+import akka.actor.{Actor, ActorLogging, Props}
 import akka.event.LoggingReceive
-import cromwell.core.JobKey
+import cromwell.backend.BackendJobDescriptorKey
+
+object JobStoreReader {
+  def props() = Props(new JobStoreReader())
+}
 
 class JobStoreReader extends Actor with ActorLogging {
   override def receive = LoggingReceive {
-    case QueryJobCompletion(jobKey) => sender ! JobNotComplete(jobKey)
+    case QueryJobCompletion(jobKey) =>
+      sender ! JobNotComplete(jobKey)
+      // TODO PBE Decide if we want to make this a singleton that stays alive or a do-and-die actor
+      // For now it kills itself after responding
+      context stop self
     case unknownMessage => log.error(s"Unexpected message to JobStoreReader: $unknownMessage")
   }
 }
 
 /**
   * Message to query the JobStoreReader, asks whether the specified job has already been completed.
   */
-case class QueryJobCompletion(jobKey: JobKey)
+case class QueryJobCompletion(jobKey: BackendJobDescriptorKey)
```

Comment:
Was there a reason this changed to BackendJobDescriptorKey? (In my change I think I modified this to just be basic types and a workflow ID because I was tired of inventing `Scopes` for the tests)